spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkWrite.java

```diff
@@ -654,6 +654,7 @@ public DataWriter<InternalRow> createWriter(int partitionId, long taskId, long e
               .dataFileFormat(format)
               .dataSchema(writeSchema)
               .dataSparkType(dsSchema)
+              .dataSortOrder(table.sortOrder())
```

Comment:
There is no guarantee that this file is actually sorted when written using the Spark Writer. I think we'll need to do a bit of extra work for this to actually be the case.

Reply:
Say I have configured sort order and I do INSERT into, can you please explain when/why the data file will not be sorted in this case?

Reply:
I have updated it to handle "write.distribution-mode" = None case.

I also observed that the testcase I added is using `range` distribution-mode.

I didn't find whether `hash` distribution-mode will use sortkey. Because table property documentation doesn't mention about it. 

> Defines distribution of write data: none: don’t shuffle rows; hash: hash distribute by partition key ; range: range distribute by partition key or sort key if table has an SortOrder

Meanwhile, I am trying to read about this. Please let me know if you know of any other cases where data won't be sorted when the sort order is configured.